contrib/wizard/src/Plugin/WizardStep/EntityFormMode.php

```diff
@@ -52,13 +54,18 @@ public function validateForm(array &$form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
+    $cached_values = $form_state->getTemporaryValue('wizard');
+    /* @var WizardInterface $config */
+    $config = $cached_values['flexiform_wizard'];
     $entity = $this->getContextValue('entity');
 
     $form_display = $form_state->get('form_display');
     $form_display->extractFormValues($entity, $form, $form_state);
 
     $form_display->formSubmitComponents($form, $form_state);
-    $entity->save();
+    if (!$config->shouldSaveOnFinish()) {
+      $entity->save();
+    }
     $form_display->saveFormEntities($form, $form_state);
```

Comment:
I think this is going to save any entities added to the step that aren't the 'base entity'. As way to test would be to add the Node Author to the first page form then see if any changes made to the author were saved.

That may be where we need to pass the alternative FormEntityManager.

Reply:
Thanks, I'll investigate